### PR TITLE
feat(mcptools): Add gen_ai.tool.call.arguments/result to the gateway's MCP tool-call span

### PR DESCRIPTION
**Pull request by SoumyaRaikwar**
## Which problem is this PR solving?
- Part of #8343 (Tracing for internal AI workflows). Follow-up to #9011: that PR covered the Python sidecar's tool-call spans, but the Go gateway's `mcptools` tracing middleware span (`tools/call <name>`) — the actual execution boundary backing both the shared MCP endpoint (external clients like Cursor/IDEs) and the turn-scoped endpoint (sidecars) — only tagged `gen_ai.tool.name`/`gen_ai.operation.name`, not what was asked or returned.

## Description of the changes
- Set `gen_ai.tool.call.arguments` and `gen_ai.tool.call.result` on the `mcptools` tracing middleware's `tools/call` span (`middleware.go`), truncated at 65536 chars (comfortably above the ~5000-token skill budget) so an oversized tool payload (e.g. a large `search_traces` dump) can't trip OTLP exporter/backend attribute-size limits.
- Added `GenAIToolCallArguments`/`GenAIToolCallResult` to `otelsemconv`, wrapping the existing semconv v1.41.0 attribute keys (no convenience `.String()` helper existed for these two yet, unlike `GenAIToolName`).
- This boundary is the more durable of the two fixes going forward: per RFC 0008 it backs both MCP mounts today and survives the sidecar migration RFC 0008 describes (M6), unlike the sidecar-side spans it duplicates for real MCP tool calls.

## How was this change tested?
- `go test ./cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/...` — 100% coverage on the changed code in `middleware.go`, including the truncation edge cases and the JSON-marshal-failure fallback.
- `go test ./internal/telemetry/otelsemconv/...` — new attribute helpers covered.
- `make fmt` / `make lint` — clean, 0 issues.
- Live E2E against a real Gemini sidecar + local Jaeger: drove a real `search_traces` tool call through the gateway's MCP endpoint and confirmed the `[jaeger] tools/call search_traces` span now carries the real arguments (`{"service_name":"jaeger","search_depth":3}`) and the real result content.

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully: `make lint test`

## AI Usage in this PR (choose one)
See [AI Usage Policy](https://github.com/jaegertracing/jaeger/blob/main/CONTRIBUTING_GUIDELINES.md#ai-usage-policy).
- [ ] **None**: No AI tools were used in creating this PR
- [ ] **Light**: AI provided minor assistance (formatting, simple suggestions)
- [x] **Moderate**: AI helped with code generation or debugging specific parts
- [ ] **Heavy**: AI generated most or all of the code changes

### CODE REVIEW

File: cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go [hunk 1]
```diff
@@ -152,6 +165,49 @@ func toolNameFromRequest(method string, req mcp.Request) string {
...
+	return string(data)
+}
+
+// truncateForSpan caps text at maxChars before it's set as a span attribute
+// value. See maxSpanAttrChars for why.
+func truncateForSpan(text string, maxChars int) string {
+	if len(text) <= maxChars {
+		return text
+	}
+	suffix := fmt.Sprintf("... [truncated, %d chars total]", len(text))
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go]: Please make the marker a prefix rather than a suffix:

```
(truncated from 2359296) {"content":[{"type":"text", ...
```

Two reasons. 64KB is already a long string, so a UI will clip or collapse the value, and a marker at the very end is the one part the reader never sees — a prefix survives every rendering, including a one-line preview. And the leading `(` means the value no longer starts with `{`, so a UI that has special rendering for `gen_ai.*` attributes will not try to parse it as JSON and then show a parse error. With the suffix form the value still opens with `{` and looks parseable, which is the worst case: it is malformed JSON that advertises itself as JSON.

File: cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go [hunk 2]
```diff
@@ -152,6 +165,49 @@ func toolNameFromRequest(method string, req mcp.Request) string {
...
+	if maxChars <= len(suffix) {
+		return suffix[:maxChars]
+	}
+	keep := maxChars - len(suffix)
+	return text[:keep] + suffix
```
Comment (yurishkuro) [inline cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go]: `text[:keep]` cuts on a byte boundary. Tool results carry arbitrary data — service names, log messages, span tags — so a multi-byte UTF-8 rune straddling `keep` is split and the attribute value becomes invalid UTF-8. Protobuf `string` fields must hold valid UTF-8 and the Go marshaler enforces it (`errInvalidUTF8` in `protobuf/internal/impl`), so this fails the OTLP export of the whole message. That is the failure mode the cap exists to prevent, so the fix must not introduce it.

The SDK's own truncation gets this right — `truncate()` in `sdk/trace/span.go` counts runes and cuts on a rune boundary. Either count runes the same way, or keep the byte cap and back `keep` off to a boundary with `utf8.DecodeLastRuneInString`. Please add a test case with a multi-byte rune landing on the cut point, asserting `utf8.ValidString` on the result.

**Review by yurishkuro — CHANGES_REQUESTED** [review]
Capping the attribute is the right call, and for the reason worth stating explicitly in the code: the SDK does not do it for you. `DefaultAttributeValueLengthLimit` is `-1` (unlimited) and nothing in Jaeger sets `SpanLimits` or `OTEL_SPAN_ATTRIBUTE_VALUE_LENGTH_LIMIT`, so an oversized tool result is exported in full, and because the batch processor packs many spans into one OTLP request, one huge value can fail the export of every span batched with it.

Two problems with how `truncateForSpan` does the capping, both inline below: the marker is in the place the reader will not see it, and the cut can produce invalid UTF-8 — which triggers the very export failure the cap is there to prevent.

---

🤖 Drafted with [Claude Code](https://claude.com/claude-code), reviewed by @yurishkuro

File: cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go (hunk 1 above)
Comment (SoumyaRaikwar) [reply cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go]: done

File: cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go (hunk 2 above)
Comment (SoumyaRaikwar) [reply cmd/jaeger/internal/extension/jaegerquery/internal/mcptools/middleware.go]: fixed, keep now backs off with `utf8.RuneStart` before slicing; added `TestTruncateForSpanCutsOnRuneBoundary` , which sweeps the cap across a three-byte rune and asserts `utf8.ValidString`.

**Review by yurishkuro — APPROVED** [review]
